CMakeLists.txt

```diff
@@ -15,6 +15,7 @@ project(ClaraGenomicsAnalysisCPP)
 option(cga_enable_tests "Build ClaraGenomicsAnalysis unit tests" OFF)
 option(cga_enable_benchmarks "Build ClaraGenomicsAnalysis benchmarks" OFF)
 option(cga_build_shared "Build ClaraGenomicsAnalysis libraries as shared objects" OFF)
+option(cga_device_synchronize_kernels "Run cudaDeviceSynchronize() in CGA_CU_CHECK_ERR calls" OFF)
```

Comment:
Ideally I wouldn't make this a separate option, but just add it the default Debug mode flags. However, I don't know how to do that (it seems cumbersome), so I guess this is the best solution.